time/src/gloo.rs

```diff
@@ -41,3 +42,7 @@ pub fn sleep(duration: Duration) -> impl Future<Output = ()> {
         .expect("Duration as millis must fit in u32");
     SendWrapper::new(TimeoutFuture::new(millis))
 }
+
+pub fn sleep_until(deadline: Instant) -> impl Future<Output = ()> {
+    sleep(deadline - Instant::now())
```

Comment:
```suggestion
    sleep(deadline.saturating_duration_since(Instant::now()))
```
Otherwise, this could panic if `deadline` is in the past.